File: 1302-DeleteCharactersToMakeFancyString/1302-DeleteCharactersToMakeFancyString.py
# Last updated: 7/31/2025, 12:28:41 PM
class Solution:
    def makeFancyString(self, s: str) -> str:
        ns=s[0]
        count=0
        for i in range(1,len(s)):
            if count>=1:
                if s[i]==s[i-1]:
                    continue
                else:
                    count=0
            if count<1:
                ns=ns+s[i]
                if s[i]==s[i-1]:
                    count+=1
                else:
                    count=0
                
        return ns   
        

File: 1302-DeleteCharactersToMakeFancyString/1302-DeleteCharactersToMakeFancyString.java
// Last updated: 7/31/2025, 12:28:36 PM
class Solution {
    public String makeFancyString(String s) {
        StringBuilder ns = new StringBuilder();
        ns.append(s.charAt(0));
        int count = 0;

        for (int i = 1; i < s.length(); i++) {
            if (count >= 1 && s.charAt(i) == s.charAt(i - 1)) {
                continue;
            } else if (s.charAt(i) == s.charAt(i - 1)) {
                count++;
            } else {
                count = 0;
            }
            ns.append(s.charAt(i));
        }

        return ns.toString();
    }
}
